joomla/joomla-cms
# [5.1] CLI: Adding command to manage Joomla core update channels

### Summary of Changes
On the CLI, we can check for updates of the core and we can run the update as well, but we actually don't know what we are updating to, what update channel we are on and what the update process is actually doing right now. This PR tries to fix some of the issues by allowing to manage the update channels.

By calling `php cli/joomla.php core:update:channel`, you get the update channel the site is currently on. By adding the parameter `default`, `next`, `testing` or `custom`, you can set the update channel to that respective one. For `custom` you need to also add the `--url=<url>` parameter for the custom URL.


### Testing Instructions
1. Run `php cli/joomla.php core:update:channel` to display the currently configured update channel.
2. Run `php cli/joomla.php core:update:channel next` to set it to the "next" update channel. Check that the params column in `#__extensions` for the `com_joomlaupdate` row contains the updated `updatesource` value and that `location` column for `Joomla! Core` in `#__update_sites` contains `https://update.joomla.org/core/sts/list_sts.xml` as URL.
3. Run it in addition with other combinations of options and see the change reflected in the core tables.
4. Run `php cli/joomla.php core:check-updates` to check for updates from the new update source.



### Link to documentations
Please select:
